### PR TITLE
Feature/detailed movie

### PR DESCRIPTION
**Pull request by pink-bunny**

### CODE REVIEW

File: state/data/reducer.js [hunk 1]
```diff
@@ -20,10 +21,13 @@ const data = (state = initialState, action) => {
         ...state,
         lists: action.lists
       };
-    case types.MERGE_LISTS_LIST:
+    case types.MERGE_MOVIES:
       return {
         ...state,
-        lists: action.lists
+        movie: {
```
Comment (plzen) [inline state/data/reducer.js]: You don't need to have separate state for movie, you already have movies - this is the entry point for all movies which used in the app

File: state/data/actions.js [hunk 2]
```diff
@@ -9,3 +9,8 @@ export const mergeListsList = (lists) => ({
   type: types.MERGE_LISTS_LIST,
   lists
 });
+
+export const mergeMovie = (movie) => ({
```
Comment (plzen) [inline state/data/actions.js]: I think you can reuse `mergeMoviesList`

File: src/pages/Movie/Carousel/component.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+            />
+          </div>
+        ))}
+      </AntdCarousel>
+    </Col>
+  </Row>
+);
+
+Carousel.propTypes = {
+  backdrops: PropTypes.arrayOf(PropTypes.object)
```
[the remark this answers is not in the document]
Comment (pink-bunny) [reply src/pages/Movie/Carousel/component.js]: In API this field is optional
https://developers.themoviedb.org/3/movies/get-movie-images
That is why we can't sure about this field.
I can check backdrops before mapping.

File: src/pages/Movie/Cast/component.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    </Row>
+  </>
+);
+
+Cast.propTypes = {
+  cast: PropTypes.arrayOf(PropTypes.object)
+};
+
+Cast.defaultProps = {
+  cast: null
```
[the remark this answers is not in the document]
Comment (pink-bunny) [reply src/pages/Movie/Cast/component.js]: All movie fields are optional. That is why added default props.

File: src/pages/Movie/component.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+          <Typography.Paragraph type="danger">
+            {error}
+          </Typography.Paragraph>
+        </Col>
+      </Row>
+    );
+  }
+
+  return (
+    details && (
```
[the remark this answers is not in the document]
Comment (pink-bunny) [reply src/pages/Movie/component.js]: I can combine loading and empty details together and here check and print sections separately

details && ()
replace to
details.backdrop && (render slider)
details.cast && (render cast)

File: state/data/reducer.js (hunk 1 above)
Comment (pink-bunny) [reply state/data/reducer.js]: Is this okay to override(if it exists) movie short movie info fetched for list preview?

File: src/pages/Movie/Carousel/component.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (pink-bunny) [reply src/pages/Movie/Carousel/component.js]: Fixed.
As we discussed API documentation is not correct and backdrops are required.

**Pull request merged**